### PR TITLE
Granular PageTypes for non-r2 Reddit apps (d2x, modmail, ads)

### PR DESCRIPTION
**Pull request by andytuba**
d2x is the "new stack" desktop site: https://www.reddit.com/user/Shitty_Watercolour

Needs testing:

* unit tests
* will modules with only "exclude" run on non-r2 pages? EDIT: floater and showimages should specifically exclude, why still running?
* will CSS targeted at r2 ("current reddit.com") leak into d2x, mod., or ad.?

Relevant issue: #4128 
Tested in browser: no tests yet

### CODE REVIEW

File: lib/utils/location.js [hunk 1]
```diff
@@ -76,12 +78,44 @@ export const execRegexes: { [k: string]: (path: string) => ?string[] } = {
...
+	}
+	return 'd2';
+});
+
+const appPageTypes: AppPageTypes = {
+	r2: {
+		default: 'linklist',
+		pageTypes: ['wiki', 'search', 'stylesheet', 'modqueue', 'subredditAbout', 'comments', 'commentsLinklist', 'profile', 'liveThread', 'inbox', 'submit', 'account', 'prefs'],
+	},
+	d2x: {
```
Comment (andytuba) [inline lib/utils/location.js]: Default : d2x

Comment (andytuba) [inline lib/utils/location.js]: added to pagetype instead

**Comment by andytuba** [comment]
Quick CSS audit:

*  🔕 selectedEntry.scss contains a bare `.entry { padding-right: 5px; }`.
* 🔕  showImage.scss conatains a bare `img { image-orientation: from-image; }
* :no_bell: res.scss contains `.md .usertext-body img { display: inherit; }`

**Comment by erikdesjardins** [comment]
Note: (#4322)

image

**Pull request merged**